### DISCUSSION TITLE
hourly error

### DISCUSSION THREAD

**Discussion by jithendra945** [cuebook/CueObserve#75] [General]
![image](https://user-images.githubusercontent.com/83579160/128318855-c74283cc-ee90-40a0-b24c-6ebb03a3a85f.png)
Im getting this error, when i am trying to run anomaly definition. 

Im not getting why it is having None in it.

**Comment by sachinkbansal** [comment]
created issue #79

**Reply by jithendra945** [reply]
I'm having 1 timestamp, 1 dimension, 1 measure
Granuality is Hour.
Issue is coming when I'm giving anomaly definition using dimension. it is showing error.

**Reply by sachinkbansal** [reply]
Can you pls share the SQL query (mask confidential table/column names, if any)?
Also, which database are you using?

**Reply by jithendra945** [reply]
big query database

SELECT
TIMESTAMP_TRUNC(created_on, HOUR) as Date,
plan,
COUNT(1) as count, -- measures
FROM table
WHERE created_on >= TIMESTAMP_SUB(TIMESTAMP_TRUNC(CURRENT_TIMESTAMP(), DAY), INTERVAL 15 DAY)
GROUP BY 1,2
ORDER BY 1,2

**Reply by jithendra945** [reply]
If u are able to reproduce the issue, 
Im  Waiting for a solution.

**Reply by jithendra945** [reply]
I tried with another table,
To reproduce :-
First I dint kept where clause, granuality is Day
and it Succeded.

Now, I kept Where Clause limiting the data to 30days and 60days, and the same Error is Coming.

**Reply by ankitkpandey** [reply]
Hi @jithendra945 , thanks for helping us find this issue. Actually, the anomaly service was returning None object in case of insufficient data. Right now it expects individual dataframes for each dimension value to meet certain conditions. As you can see, we have currently hardcoded the df to have atleast 20 datapoints for prophet needs it to have a meaningful trend analysis.
```
def dataFrameEmpty(df):
    """Checks whether dataFrame has enough data for prophet"""
    if df is None:
        return True
    if df.empty:
        return True
    if df.shape[0] < 20:
        return True
    return False
```

**Reply by ankitkpandey** [reply]
I'm currently in the process of handling it as an error and logging it as Insufficient data, later we could maybe make the hardcoded 20 rows condition variable. 
Anyway, thanks again. I'd recommend running it on data with more than 20 days of data for each dimension in the meantime

**Comment by jithendra945** [comment]
For each dimension, there should be at least 20 days of Data.

Got it Ankit.
Thanks. I'm facing this issue for a couple of days.

If I will check before giving the query, then it shouldn't be a problem
Right. ?

**Reply by ankitkpandey** [reply]
Yess @jithendra945 , It should work in that case. Also I will raise a PR by tonight which handles the case for data less than 20 days and adds it to the lag instead of returning None
